cider/exec/nextgen/parsers/RelAlgEUParser.cpp

```diff
@@ -0,0 +1,147 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "exec/nextgen/parsers/Parser.h"
+
+#include "exec/nextgen/operators/FilterNode.h"
+#include "exec/nextgen/operators/ProjectNode.h"
+#include "exec/nextgen/operators/SourceNode.h"
+#include "util/Logger.h"
+
+namespace cider::exec::nextgen::parsers {
+using namespace cider::exec::nextgen::operators;
+
+static bool isParseable(const RelAlgExecutionUnit& eu) {
+  if (!eu.join_quals.empty()) {
+    LOG(ERROR) << "JOIN is not supported in RelAlgExecutionUnitParser.";
+    return false;
+  }
+  if (eu.groupby_exprs.empty()) {
+    LOG(ERROR) << "Aggregation is not supported in RelAlgExecutionUnitParser.";
+    return false;
+  }
+  if (eu.groupby_exprs.size() > 1 || *eu.groupby_exprs.begin() != nullptr) {
+    LOG(ERROR) << "GroupBy is not supported in RelAlgExecutionUnitParser.";
+    return false;
+  }
+
+  return true;
+}
+
+// Used for input columnVar collection and combination
+class InputAnalyzer {
+ public:
+  InputAnalyzer(const std::vector<InputColDescriptor>& input_desc, OpPipeline& pipeline)
+      : input_desc_(input_desc)
+      , pipeline_(pipeline)
+      , input_exprs_(input_desc.size(), nullptr) {}
+
+  void run() {
+    if (pipeline_.empty()) {
+      return;
+    }
+
+    for (size_t i = 0; i < input_desc_.size(); ++i) {
+      input_desc_to_index_.insert({input_desc_[i], i});
+    }
+
+    for (auto& op : pipeline_) {
+      auto exprs = op->getExprs();
+      for (auto& expr : exprs) {
+        traverse(&expr);
+      }
+    }
+
+    for (size_t i = 0; i < input_exprs_.size(); ++i) {
+      if (nullptr == input_exprs_[i]) {
+        LOG(FATAL) << "Input column expr missed. Column descriptor = "
+                   << input_desc_[i].toString();
+      }
+    }
+
+    pipeline_.insert(pipeline_.begin(), createOpNode<SourceNode>(input_exprs_));
+  }
+
+ private:
+  void traverse(ExprPtr* curr) {
+    if (auto col_var_ptr = dynamic_cast<Analyzer::ColumnVar*>(curr->get()); col_var_ptr) {
+      auto iter = input_desc_to_index_.find(
+          InputColDescriptor(col_var_ptr->get_column_info(), col_var_ptr->get_rte_idx()));
+      CHECK(iter != input_desc_to_index_.end());
+
+      size_t index = iter->second;
+      if (input_exprs_[index]) {
+        *curr = input_exprs_[index];
```

Comment:
`ColumnVar` dedup?

Reply:
Replace a duplicated ```ColumnVar``` with the first founded one.

Reply:
Why need to do this?

Reply:
> Why need to do this?

To make input columns are decoded only once.